staking-pool-shares/src/lib.rs

```diff
@@ -0,0 +1,759 @@
+            self.update_context(staking(), 0);
+            let total_stake = self.contract.total_staked_balance;
+            // First function call action
+            self.contract.ping();
+            // Stake action
+            self.amount = self.amount + self.locked_amount - total_stake;
+            self.locked_amount = total_stake;
+            // Second function call action
+            self.update_context(staking(), 0);
+            self.contract.internal_after_stake();
+        }
+
+        pub fn skip_epochs(&mut self, num: EpochHeight) {
+            self.epoch_height += num;
+            self.locked_amount = (self.locked_amount * (100 + u128::from(num))) / 100;
+        }
+    }
+
+    #[test]
+    fn test_deposit_withdraw() {
+        let mut emulator = Emulator::new(
+            owner(),
+            "KuTCtARNzxZQ3YvXDeLjx83FDqxv2SdQTSbiq876zR7".to_string(),
+            zero_fee(),
+        );
+        let deposit_amount = ntoy(1_000_000);
+        emulator.update_context(bob(), deposit_amount);
+        emulator.contract.deposit();
+        emulator.amount += deposit_amount;
+        emulator.update_context(bob(), 0);
+        assert_eq!(
+            emulator.contract.get_account_unstaked_balance(bob()).0,
+            deposit_amount
+        );
+        emulator.contract.withdraw(deposit_amount.into());
+        assert_eq!(
+            emulator.contract.get_account_unstaked_balance(bob()).0,
+            0u128
+        );
+    }
+
+    #[test]
+    fn test_stake_with_fee() {
+        let mut emulator = Emulator::new(
+            owner(),
+            "KuTCtARNzxZQ3YvXDeLjx83FDqxv2SdQTSbiq876zR7".to_string(),
+            RewardFeeFraction {
+                numerator: 10,
+                denominator: 100,
+            },
+        );
+        let deposit_amount = ntoy(1_000_000);
+        emulator.update_context(bob(), deposit_amount);
+        emulator.contract.deposit();
+        emulator.amount += deposit_amount;
+        emulator.update_context(bob(), 0);
+        emulator.contract.stake(deposit_amount.into());
+        emulator.simulate_stake_call();
+        assert_eq!(
+            emulator.contract.get_account_staked_balance(bob()).0,
+            deposit_amount
+        );
+
+        emulator.skip_epochs(10);
+        // Overriding rewards (+ 100K reward)
+        emulator.locked_amount = deposit_amount + ntoy(100_000);
+        emulator.update_context(bob(), 0);
+        emulator.contract.ping();
+        assert_eq_in_near!(
+            emulator.contract.get_account_staked_balance(bob()).0,
+            deposit_amount + ntoy(90_000)
+        );
+        // Owner got 10% of the rewards
+        assert_eq_in_near!(
+            emulator.contract.get_account_staked_balance(owner()).0,
+            ntoy(10_000)
+        );
+
+        emulator.skip_epochs(10);
+        // Overriding rewards (another 100K reward)
+        emulator.locked_amount = deposit_amount + ntoy(100_000) + ntoy(100_000);
```

Comment:
In skip_epochs you calculate 
`self.locked_amount = (self.locked_amount * (100 + u128::from(num))) / 100;`

But then it's overrides immediately. Whats' a point in this calculation?

Reply:
To skip 10 epochs